core/eolearn/core/eodata.py

```diff
@@ -162,6 +191,25 @@ def _parse_feature_value(self, value: object, feature_name: str) -> np.ndarray:
 
         return value
 
+    def _update_temporal_dim(self, new_value: int | None) -> None:
+        self._temporal_dim = new_value
+        if self._temporal_dim is None:
+            if self._content:
+                warnings.warn(
+                    f"EOPatch does not have timestamps, but has temporal features of type {self.feature_type}.",
+                    category=TemporalDimensionWarning,
+                    stacklevel=4,
+                )
+            return
+
+        for feature_name, value in self._content.items():
+            if not isinstance(value, FeatureIO) and value.shape[0] != self._temporal_dim:
```

Comment:
What else can it also be, if not `isinstance(value, FeatureIO)`?

Reply:
it can be an already loaded value. Content has both `FeatureIO[T]` and `T` values (if something is lazy loaded the FeatureIO is replaced with the value itself)